### PR TITLE
Fix potential infinity loop.

### PR DESCRIPTION
**Pull request by raniervf**
for loop is idiom to interate with clists.

### CODE REVIEW

File: src/driver/interface/mailstorage.c [hunk 1]
```diff
@@ -320,7 +320,7 @@ void mailstorage_disconnect(struct mailstorage * storage)
 {
   clistiter * cur;
 
-  while ((cur = clist_begin(storage->sto_shared_folders)) != NULL) {
+  for (cur = clist_begin(storage->sto_shared_folders); cur != NULL; cur = clist_next(cur)) {
```
Comment (dinhviethoa) [inline src/driver/interface/mailstorage.c:323]: mailfolder_disconnect() is probably touching the data structure storage->sto_shared_folders... the new version might lead to some inconsistencies / crashes.

**Comment by dinhviethoa** [comment]
We'd rather drop that pull request.

File: src/driver/interface/mailstorage.c (hunk 1 above)
Comment (raniervf) [reply src/driver/interface/mailstorage.c:323]: Well, the alert is the loop break don't depend interations.
I think the loop can run forever as it is written.

**Comment by raniervf** [comment]
>We'd rather drop that pull request.
No problem, I understand.

**Pull request closed without merging**